src/pybind/mgr/rbd_support/schedule.py

```diff
@@ -343,8 +343,20 @@ def __len__(self):
 
     def load(self, namespace_validator=None, image_validator=None):
 
-        schedule_cfg = self.handler.module.get_localized_module_option(
+        schedule_cfg = self.handler.module.get_module_option(
             self.handler.MODULE_OPTION_NAME, '')
+
+        # Previous versions incorrectly stored the global config in
+        # the localized module option. Check the config is here and fix it.
+        if not schedule_cfg:
+            schedule_cfg = self.handler.module.get_localized_module_option(
+                self.handler.MODULE_OPTION_NAME, '')
+            if schedule_cfg:
+                self.handler.module.set_module_option(
+                    self.handler.MODULE_OPTION_NAME, schedule_cfg)
+        self.handler.module.set_localized_module_option(
```

Comment:
Nit: perhaps put this under the `if` clause at line 354 after moving it to the non-localized module options. There is technically a possibility of a MGR crash that will leave the old config under the localized section, but it seems unlikely enough and eventually we can remove all this code.

Reply:
I intentionally put this out of the `if` clause, so it always had a chance to clean the old option. Consider a case when an old localized option exists for both mgr.x and mgr.y (`mgr/rbd_support/x/mirror_snapshot_schedule` and `mgr/rbd_support/y/mirror_snapshot_schedule`). After upgrade the active mgr.x "fixes" `mgr/rbd_support/x/mirror_snapshot_schedule` making it `mgr/rbd_support/mirror_snapshot_schedule`. Then mgr.y becomes active, and if it is under `if` clause it will not remove `mgr/rbd_support/y/mirror_snapshot_schedule` because `mgr/rbd_support/mirror_snapshot_schedule` exists. And then if the user decides to remove the global schedule (`mgr/rbd_support/mirror_snapshot_schedule`) the mgr.y will mysteriously reuse `mgr/rbd_support/y/mirror_snapshot_schedule`.